cmd/thv-operator/DESIGN.md

```diff
@@ -1,44 +1,139 @@
 # Design & Decisions
 
-This document aims to help fill in gaps of any decision that are made around the design of the ToolHive Operator.
+This document captures architectural decisions and design patterns for the ToolHive Operator. For user-facing documentation, see [REGISTRY.md](REGISTRY.md).
```

Comment:
Should this link be in the main thv-operator/design.md? Registry.md is very registry-specific

Reply:
Removed. The link remains in `cmd/thv-operator/README.md`, more focused on user-facing documentation